### DISCUSSION TITLE
Wallet can’t connect (sync) to FullNode on local network

### DISCUSSION THREAD

**Discussion by Oscillator098** [Chia-Network/chia-blockchain#8036] [Support]
I'm lost and I don't know what else to do but cry here :slight_smile: 

Setup:
- PC1 (host, Windows 10, Chia 1.2.3.): **192.168.1.2** FullNode, Wallet1 key, all synced, everything working perfectly 
- PC2 (VirtualMachine (same hardware PC1),Windows 10, Chia 1.2.3.): **192.168.2.3**., Wallet2 key, No FullNode, no Harvester, no Farmer, running only Wallet and daemon.
- PC1 can ping PC2 and vice-versa, both PCs are connected to internet (shared vEthernet adapter on host PC1).
- Both machines have synced Time and TimeZone from windows time server.
- I tried also turning off Firewalls on both machines.

Same exact setup I have on Flax VirtualMachine and it is working perfectly and I have no clue what is different.

- PC2 config file: 
```
...
wallet:
  ...
  full_node_peer:
    host: 192.168.1.2
    port: 8444
...
```

Without this config edit PC2 can start sync normally over internet's FullNode and Wallet2 is working.
When I apply this config edit and I want my PC1 to be FullNode for PC2 then Wallet sync issues appear.

PC1 sees that PC2 has connected its wallet to PC1. PC1 starts sending some data and then it stops.
![image](https://user-images.githubusercontent.com/47561865/129154514-1a1c294c-9822-4f79-b2b0-a50e918c9a2a.png)

**This is from my log on PC2:**
```
wallet chia.wallet.wallet_node    : INFO     Loop end in sync job
wallet chia.wallet.wallet_node    : INFO     Loop start in sync job
wallet wallet                     : INFO     Reconnecting to peer {'host': '192.168.1.2', 'port': 8444}
wallet wallet_server              : INFO     Connected with full_node {'host': '192.168.1.2', 'port': 8444}
wallet chia.wallet.wallet_node    : INFO     self.sync_event.set()
wallet chia.wallet.wallet_node    : INFO     Have collected 1 potential peaks
wallet chia.wallet.wallet_node    : INFO     Requesting blocks 0-32
wallet wallet_server              : INFO     Connection closed: 192.168.1.2, node id: 
wallet chia.wallet.wallet_node    : ERROR    Error while trying to fetch from peer:Peer returned no response Traceback (most recent call last):
  File "chia\wallet\wallet_node.py", line 527, in batch_sync_to_peak
  File "chia\wallet\wallet_node.py", line 679, in fetch_blocks_and_validate
ValueError: Peer returned no response

wallet chia.wallet.wallet_node    : ERROR    Loop exception in sync Was not able to add blocks 0-32. Traceback (most recent call last):
  File "chia\wallet\wallet_node.py", line 580, in sync_job
  File "chia\wallet\wallet_node.py", line 660, in _sync
  File "chia\wallet\wallet_node.py", line 537, in batch_sync_to_peak
RuntimeError: Was not able to add blocks 0-32

```

In DEBUG mode I get:
`Time for request request_header_blocks: {'host': '192.168.1.2', 'port': 8444} = 25.126431465148926, None? True`
And this keeps repeating.

Additional info for VM, maybe there is something:
PC1:
LAN: IP 192.168.1.2, Gate 192.168.1.1 (home ISP router) , Subnet 255.255.255.0, DNS 8.8.8.8 and 8.8.4.4 (internet Sharing to virtualAdapter)
virtualEthernet adapter: IP 192.168.2.4, Gate 192.168.1.2, Subnet 255.255.0.0, DNS none

PC2 (VM):
LAN: IP 192.168.2.3, Gate 192.168.2.4, Subnet 255.255.255.0, DNS 8.8.8.8 and 8.8.4.4

With this config I have internet and ping in both directions. I’m questioning maybe Subnet setup on virtualEthernet on PC1. Should it be 255.255.240.0 or something

What am I missing? What else can I check?

**Comment by keliew** [comment]
https://github.com/Chia-Network/chia-blockchain/wiki/Windows-Tips-and-Tricks#4-disable-upnp-when-running-multiple-nodes
https://github.com/Chia-Network/chia-blockchain/wiki/FAQ#full-node
https://github.com/Chia-Network/chia-blockchain/wiki/Resolving-Sync-Issues---Port-8444#why-forward-port-8444
